genesyscloud/user/resource_genesyscloud_user_utils.go

```diff
@@ -287,6 +292,28 @@ func updateUserRoutingUtilization(d *schema.ResourceData, proxy *userProxy) diag
 	return nil
 }
 
+func updatePassword(ctx context.Context, d *schema.ResourceData, proxy *userProxy) diag.Diagnostics {
+	if !d.HasChange("password") {
+		return nil
+	}
+
+	password, ok := d.Get("password").(string)
+	if !ok {
+		return util.BuildDiagnosticError(ResourceType, "Invalid password type", fmt.Errorf("password must be a string"))
+	}
```

Comment:
I think ok will be false here if password is an empty string. Terraform will fail right away if the wrong type is assigned to a field.

Reply:
If password is not provided by the user, then requesting the second return value (`ok`) will ensure that `password` is the zero value for whatever type we are casting to right of `:=`

I know this is how it works when casting map items... pretty sure its the same casting things returned from `d.Get`

Reply:
Ah. Are you thinking we just remove these extra checks then and let the API handle the password update straight away?